code/datums/uplinks_items.dm

```diff
@@ -1034,3 +1053,29 @@
 		return new I.item(loc)
 	else
 		to_chat(user, "<span class='warning'>There is no available items you could buy for [U.uses] TK.</span>")
+
+/datum/uplink_item/badass/surplus_crate/spawn_item(turf/loc, obj/item/device/uplink/U)
+	var/obj/structure/closet/crate/C = new(loc)
+	var/list/temp_uplink_list = get_uplink_items(U)
+	var/list/buyable_items = list()
+	for(var/category in temp_uplink_list)
+		buyable_items += temp_uplink_list[category]
+	var/remaining_TC = crate_value
+	var/list/bought_items = list()
+	var/list/itemlog = list()
+	U.uses -= cost
+
+	var/datum/uplink_item/I
+	while(remaining_TC)
+		I = pick(buyable_items)
+		if(I.cost > remaining_TC)
+			continue
+		if((I.item in bought_items) && prob(33)) //To prevent people from being flooded with the same thing over and over again.
+			continue
+		bought_items += I.item
```

Comment:
```suggestion
		if((I.type in bought_items) && prob(33)) //To prevent people from being flooded with the same thing over and over again.
			continue
		bought_items += I.type
```
думаю, корректней хранить тип датума-обёртки, чем сам предмет для спавна. Могут же быть случаи, когда покупается одинаковый предмет, но там с разными приколами?

Reply:
поч резолвнул, но не сделал?

Reply:
так не работает, предметы не вызываются

Reply:
ты прав